keyserver/server.go

```diff
@@ -22,21 +22,42 @@ import (
 	"github.com/coniks-sys/coniks-go/utils"
 )
 
+// A ServerConfig contains configuration values
+// which are read at initialization time from
+// a TOML format configuration file.
 type ServerConfig struct {
-	configFilePath      string
-	DatabasePath        string          `toml:"database"`
-	LoadedHistoryLength uint64          `toml:"loaded_history_length"`
-	Policies            *ServerPolicies `toml:"policies"`
-	Addresses           []*Address      `toml:"addresses"`
+	// DatabasePath is a path to the database for storing the server's directory.
+	DatabasePath string `toml:"database"`
+	// LoadedHistoryLength is the maximum length of current snapshots stored in memory.
+	LoadedHistoryLength uint64 `toml:"loaded_history_length"`
+	// Policies contains the server's CONIKS policies configuration.
+	Policies *ServerPolicies `toml:"policies"`
+	// Addresses contains the server's connections configuration.
+	Addresses      []*Address `toml:"addresses"`
+	configFilePath string
 }
 
+// An Address declares a server's connection.
+// It makes the server connections configurable
+// so that a key server implementation can easily
+// be used as a first-party identity provider or
+// a third-party communication service.
+// It supports two types of connections: a TCP connection ("tcp")
+// and a Unix socket connection ("unix"). Allowing registration
+// has to be explicitly defined for each connection.
+// Other types of requests are allowed by default.
```

Comment:
I would add the explanation here that a registration is considered a "write", while the other requests are "reads", just to make it more clear why we treat registration specially.